homeassistant/components/climate/melissa.py

```diff
@@ -0,0 +1,248 @@
+
+    add_devices(all_devices)
+
+
+class MelissaClimate(ClimateDevice):
+    """Representation of a Melissa Climate device."""
+
+    def __init__(self, api, serial_number, init_data):
+        """Initialize the climate device."""
+        self._name = init_data['name']
+        self._api = api
+        self._serial_number = serial_number
+        self._data = init_data['controller_log']
+        self._state = None
+        self._cur_settings = None
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def is_on(self):
+        """Return current state."""
+        if self._cur_settings is not None:
+            return self._cur_settings[self._api.STATE] in (
+                self._api.STATE_ON, self._api.STATE_IDLE)
+        return STATE_UNKNOWN
+
+    @property
+    def current_fan_mode(self):
+        """Return the current fan mode."""
+        if self._cur_settings is not None:
+            return self.melissa_fan_to_hass(
+                self._cur_settings[self._api.FAN])
+        return STATE_UNKNOWN
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._data[self._api.TEMP]
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return PRECISION_WHOLE
+
+    @property
+    def current_operation(self):
+        """Return the current operation mode."""
+        if self._cur_settings is not None:
+            return self.melissa_op_to_hass(
+                self._cur_settings[self._api.MODE])
+        return STATE_UNKNOWN
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [
+            STATE_AUTO, STATE_HEAT, STATE_COOL, STATE_DRY, STATE_FAN_ONLY
+        ]
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return [
+            STATE_AUTO, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH
+        ]
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self._cur_settings is None:
+            return None
+        return self._cur_settings[self._api.TEMP]
+
+    @property
+    def state(self):
+        """Return current state."""
+        if self._cur_settings is None:
+            return STATE_UNKNOWN
+        return self.melissa_state_to_hass(
+            self._cur_settings[self._api.STATE])
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature for the thermostat."""
+        return 16
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature for the thermostat."""
+        return 30
```

Comment:
Are the limits correct, they feel quite strict :-)

Reply:
Yes, this is strange but this is the supported max :D 
```
temp | integer($integer32)AC temperature from 16 to 30
```